src/snowflake/cli/_plugins/nativeapp/project_model.py

```diff
@@ -139,10 +136,13 @@ def project_identifier(self) -> str:
     @property
     def package_name(self) -> str:
         if self.definition.package and self.definition.package.name:
-            name = self.definition.package.name
+            return self.definition.package.name
         else:
+            # V1.0 PDF doesn't support templating, so if the identifier isn't
+            # explicitly specified, the default is generated here,
+            # so we have to append the test resource suffix here
             name = default_app_package(self.project_identifier)
```

Comment:
we need to_identifier here as well, because we don't use concat_identifiers() all the time anymore.

Reply:
`append_test_resource_suffix` now does `to_identifier` when it no-ops to make sure it always generates a valid identifier

Reply:
there is no path that doesn't go through `to_identifier` in this logic, you lost me.

Reply:
custom validator uses the append_test_resource_suffix, which would turn 'a.b' application name to '"a.b"' application name even if we are not in test mode (env variable for SUFFIX is not set)

Reply:
What's the possible negative side-effect? I thought we always have to quote the app/pkg names on use anyway.

Reply:
```
native_app:
  package:
    name: my.package
  application:
    name: <% ctx.native_app.package.name %>_app
```

app name will then turn out as '"my.package"_app'. 
I know they can use id_concat, but it's counter intuitive. It's ok to do it to defaults, but not when it's specified directly in pdf

Reply:
yeah ok that's fair, thanks for explaining. @sfc-gh-fcampbell sounds like a regression we need to address

Reply:
ah good point, i'll rework the appending part